src/overlays/actors/ovl_Item_Shield/z_item_shield.c

```diff
@@ -49,21 +40,192 @@ static ColliderCylinderInit D_80B871A0 = {
     },
     { 15, 15, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Shield/func_80B86920.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Shield/ItemShield_Init.s")
+const ActorInit Item_Shield_InitVars = {
+    ACTOR_ITEM_SHIELD,
+    ACTORCAT_ITEMACTION,
+    FLAGS,
+    OBJECT_LINK_CHILD,
+    sizeof(ItemShield),
+    (ActorFunc)ItemShield_Init,
+    (ActorFunc)ItemShield_Destroy,
+    (ActorFunc)ItemShield_Update,
+    (ActorFunc)ItemShield_Draw,
+};
+
+static Color_RGBA8 unused = { 255, 255, 0, 255 };
+static Color_RGBA8 unused2 = { 255, 0, 0, 255 };
+
+void func_80B86920(ItemShield* this, ItemShieldActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ItemShield_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ItemShield* this = THIS;
+    s32 i;
+
+    this->timer = 0;
+    this->unk_19C = 0;
+
+    switch (this->actor.params) {
+        case 0:
+            ActorShape_Init(&this->actor.shape, 1400.0f, NULL, 0.0f);
+            this->actor.shape.rot.x = 0x4000;
+            func_80B86920(this, func_80B86BC8);
+            break;
+
+        case 1:
+            ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+            func_80B86920(this, func_80B86F68);
+            this->unk_19C |= 2;
+            for (i = 0; i < 8; i++) {
+                this->unk_19E[i] = 1 + 2 * i;
+                this->unk_1A8[i].x = Rand_CenteredFloat(10.0f);
+                this->unk_1A8[i].y = Rand_CenteredFloat(10.0f);
+                this->unk_1A8[i].z = Rand_CenteredFloat(10.0f);
+            }
+            break;
+    }
+
+    Actor_SetScale(&this->actor, 0.01f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    osSyncPrintf(VT_SGR("32") "Item_Shild %d \n" VT_RST, this->actor.params);
```

Comment:
this vt macro should be a color. vt_fmt.py in tools will take the commands as a string
this one is `VT_FGCOL(GREEN)`